suite/src/benchmark.rs

```diff
@@ -0,0 +1,75 @@
+use core::arch::asm;
+
+/// Returns the machine cycle counter
```

Comment:
Did you come up with this on your own? If not, can you cite the source? (a link, or a page number in the manual etc)